### PR TITLE
redesign

### PR DESCRIPTION
**Pull request by vsemp**
Just for mentors to review it.

This is the new proposed design. I added two abstract classes: inventory_service and hardware_service.

### CODE REVIEW

File: lib/hil/hil.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    if len(sys.argv) < 2:
+        sys.exit("Should be at least one argument.")
+    else:
+        if len(sys.argv) == 2:
+            globals()[sys.argv[1]]()
+        else:
+            globals()[sys.argv[1]](sys.argv[2:])
+        
+if __name__ == "__main__":
+   main()
```
Comment (saacg) [inline lib/hil/hil.py]: please see conversation here: https://github.com/BUEC528-HIL-QUADS/quads/pull/35#discussion_r110508083

Comment (saacg) [inline lib/hil/hil.py:118]: This file provides many useful wrapper functions to abstract the rest calls to the HIL api. However, i'm not sure we want to add another cli to QUADS, let alone one that is specific to HIL. Maybe we could get rid of the command line parsing, and add these as private methods when we implement the concrete HIL inventory class?

File: lib/hil/hil_starter.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    haas project_create cloud02
+    haas network_create_simple cloud01 cloud01
+    haas network_create_simple cloud02 cloud02
+    haas project_connect_node cloud01 host01.com
+    haas node_connect_network host01.com nic cloud01 null
+}
+start_hil_data &
+sleep 1
+echo HIL data was added.
+sleep 1
```
Comment (saacg) [inline lib/hil/hil_starter.sh:51]: This script seems to require the existence of HIL in some directory on the user's local machine. I think in previous discussions it was decided that we should keep this out of QUADS? Perhaps if we want to use this we could put this in HIL instead? Then you could still use this script to automate running the servers, but would not be adding any HIL dependencies to QUADS.

File: lib/hil/hil.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+url: http://127.0.0.1:5000
```
Comment (portante) [inline lib/hil/hil.yml:1]: What is this file for exactly?  It has one line, and no comments.

**Review by portante — CHANGES_REQUESTED** [review]
It is important when writing new code and proposing changes to a body of work to properly motivate why you are making the changes.  The changes themselves are not the motivating factor.

Having a write-up in the commit message with they why, how it is of benefit to the community of users of this code base, and how you are going to use them is pretty important.

File: lib/hil/hil.yml (hunk 3 above)
Comment (vsemp) [reply lib/hil/hil.yml:1]: I'm confused. I thought commit messages are very short.

Should I put this explanation into PR comment?

**Comment by saacg** [comment]
closing as it has been replaced with newer changes.

**Pull request closed without merging**